packages/plugins/plugin-advertising-id/android/src/main/java/com/reactnativeanalyticsreactnativepluginadvertisingid/AnalyticsReactNativePluginAdvertisingIdModule.kt

```diff
@@ -0,0 +1,41 @@
+package com.analyticsreactnativepluginadvertisingid
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.Promise
+import com.facebook.react.ReactApplication
+import com.google.android.gms.ads.identifier.AdvertisingIdClient
+import com.sovranreactnative.SovranModule
+import com.facebook.react.module.annotations.ReactModule
+import android.util.Log
+
+@ReactModule(name="AnalyticsReactNativePluginAdvertisingId")
+class AnalyticsReactNativePluginAdvertisingIdModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+  override fun getName(): String {
+      return "AnalyticsReactNativePluginAdvertisingId"
+  }
+
+  @ReactMethod
+  fun getAdvertisingId(promise: Promise) {
+    val reactContext = (currentActivity?.application as ReactApplication)
+    ?.reactNativeHost
+    ?.reactInstanceManager
+    ?.currentReactContext
+
+    if (reactContext == null) {
+      promise.resolve(null)
+      return
+    }
+
+     val advertisingInfo = AdvertisingIdClient.getAdvertisingIdInfo(reactContext)
```

Comment:
Per the docs:

> Cases where Google Play Services is unavailable are indicated by a [GooglePlayServicesNotAvailableException](https://developers.google.com/android/reference/com/google/android/gms/common/GooglePlayServicesNotAvailableException) being thrown by getAdvertisingIdInfo().

We should handle exceptions here, log it and return null in those cases